manifests/plugin/remote_execution/ssh/params.pp

```diff
@@ -0,0 +1,10 @@
+# Remote Execution SSH default parameters
+class foreman_proxy::plugin::remote_execution::ssh::params {
+  $enabled           = true
+  $listen_on         = 'https'
+  $generate_keys     = true
+  $ssh_identity_dir  = "${::foreman_proxy::dir}/.ssh"
```

Comment:
I don't think (3) is possible when using strict variables, it'd throw an error.  I'd suggest just sticking with (2) and foreman_proxy::params - I can't think why you'd really ever change foreman_proxy::dir away from the default.  If you did, overriding this too isn't a big problem.

I'm sure it's more useful to override ssh_identity_dir, which (2) still allows easily.